### PR TITLE
Upgrade min SDK and web-angular

### LINKED ISSUE #506: Remove pinning of angular_compiler
- [ ] Roll angular to a version w/ good pinning (https://github.com/dart-lang/angular/issues/1325)
- [ ] Get a version of angular components that is compatible with it

### PR DESCRIPTION
**Pull request by chalin**
Fixes #506, #521, #522

### CODE REVIEW

File: templates/web-angular/pubspec.yaml [hunk 1]
```diff
@@ -5,20 +5,20 @@ description: A web app that uses AngularDart Components
...
-  angular: 5.0.0-alpha+12
-  angular_components: 0.9.0-alpha+12
+  angular: ^5.0.0-alpha+13
+  angular_components: ^0.9.0-alpha+13
 
 dev_dependencies:
-  angular_test: ^2.0.0-alpha
-  build_runner: ^0.8.7
-  build_test: ^0.10.0
+  angular_test: ^2.0.0-alpha+11
```
Comment (kevmoo) [inline templates/web-angular/pubspec.yaml]: should this sync to +13, too?

File: templates/web-angular/test/app_test.dart [hunk 2]
```diff
@@ -25,4 +25,6 @@ void main() {
   test('heading', () {
     expect(appPO.title, contains('My First AngularDart App'));
   });
+
+  // Testing info: https://webdev.dartlang.org/angular/guide/testing
```
Comment (kevmoo) [inline templates/web-angular/test/app_test.dart]: Put this on main? or the top of the file?

More text? "To learn more about testing angular apps, seee..."

Comment (chalin) [reply templates/web-angular/test/app_test.dart]: This is what we had before (and is something that @kwalrath had approved I believe. It is in the same vein as when we write simply: "More information: ..." (used in at least a dozen places on the webdev docs).

Shall we keep this for now and revisit later as needed?

Comment (kevmoo) [reply templates/web-angular/test/app_test.dart]: Can leave for now. I'd be nice if this was a bit more conspicuous so we help users get started.

File: templates/web-angular/pubspec.yaml (hunk 1 above)
Comment (kevmoo) [reply templates/web-angular/pubspec.yaml]: ack

**Comment by kevmoo** [comment]
@chalin – ship it!

**Comment by chalin** [comment]
@kevmoo - will you be publishing a new version? If so, have you decided if you really want to bump up the version to 3.0.0 (or actually bend the semver rules for this case and keep it at 2.1.0)?

**Comment by kevmoo** [comment]
> have you decided if you really want to bump up the version to 3.0.0 (or actually bend the semver rules for this case and keep it at 2.1.0)?

Let's keep w/ semver. Most users don't think about the version of this anyway. They just run 'pub global activate' (or IDEA does for them)

**Comment by chalin** [comment]
> Let's keep w/ semver

SGTM

I'm not a stagehand uploader. Will you be publishing or do you want to wait?

**Comment by kevmoo** [comment]
I can. Although look at the issue I just filed...

**Pull request merged**